src/webcomponents/spreadsheet/SpreadsheetRender.tsx

```diff
@@ -0,0 +1,15 @@
+import React, { FC } from "react";
+import { SpreadsheetRenderType } from "../../@types/interfaces";
+
+const SpreadsheetRender: FC<SpreadsheetRenderType> = (props) => {
+  return (
+    <div>
+      <div ref={props.jRef} />
+      <br />
+      <input type="button" onClick={props.addRow} value="Add new row" />
+      <input type="button" onClick={props.downloadSheet} value="Download" />
```

Comment:
Please check also the issue / requirements whether all of them are fulfilled:
1. onchange returns modified copy of languages input (functional requirement that is necessary) - in other words add button that maps spreadsheet back to JSON. You can keep also the extra button `jRef.current.jexcel.download();`.
2. implement as webcomponent (technical requirement that can be done at the end of the task once all previous comments are implemented)

Reply:
are you talking about this webcomponent ? https://developer.mozilla.org/en-US/docs/Web/Web_Components

Reply:
I pushed. can you pleaes check and let me know